### DISCUSSION TITLE
Build problem regarding snabbdom/build/jsx.js

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#1221] [Q&A] [answered]
Hi, 
I have a strange problem when I try to switch to the new eclipse/glsp-client release "2.1.0".

My app module does no longer build and I get the following compiler error message:

```
open-bpmn-app: ERROR in ../node_modules/snabbdom/build/jsx.js 2:0-24
open-bpmn-app: Module not found: Error: Can't resolve './h' in '/home/rsoika/git/open-bpmn/open-bpmn.glsp-client/node_modules/snabbdom/build'
open-bpmn-app: Did you mean 'h.js'?
open-bpmn-app: BREAKING CHANGE: The request './h' failed to resolve only because it was resolved as fully specified
open-bpmn-app: (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
open-bpmn-app: The extension in the request is mandatory for it to be fully specified.
open-bpmn-app: Add the extension to the request.
open-bpmn-app: resolve './h' in '/home/rsoika/git/open-bpmn/open-bpmn.glsp-client/node_modules/snabbdom/build'
open-bpmn-app:   using description file: /home/rsoika/git/open-bpmn/open-bpmn.glsp-client/node_modules/snabbdom/package.json (relative path: ./build)
open-bpmn-app:     Field 'browser' doesn't contain a valid alias configuration
open-bpmn-app:     using description file: /home/rsoika/git/open-bpmn/open-bpmn.glsp-client/node_modules/snabbdom/package.json (relative path: ./build/h)
open-bpmn-app:       Field 'browser' doesn't contain a valid alias configuration
open-bpmn-app:       /home/rsoika/git/open-bpmn/open-bpmn.glsp-client/node_modules/snabbdom/build/h doesn't exist
open-bpmn-app:  @ ../node_modules/snabbdom/build/index.js 20:0-41 20:0-41 20:0-41
open-bpmn-app:  @ ../node_modules/@eclipse-glsp/client/lib/views/glsp-projection-view.js 30:19-38
open-bpmn-app:  @ ../node_modules/@eclipse-glsp/client/lib/views/index.js 35:13-46
open-bpmn-app:  @ ../node_modules/@eclipse-glsp/client/lib/index.js 39:13-31
open-bpmn-app:  @ ../node_modules/@eclipse-glsp/theia-integration/lib/browser/theia-integration-frontend-module.js 18:17-48
open-bpmn-app:  @ ./src-gen/frontend/index.js 79:30-117
open-bpmn-app: webpack 5.90.0 compiled with 1 error in 11889 ms
```

I think my configuration/setup is equal to the latest [examples/workflow-theia](https://github.com/eclipse-glsp/glsp-theia-integration/tree/master/examples/workflow-theia) 

And I only switched from 

`"@eclipse-glsp/client": "2.1.0-next.303"   ==> "@eclipse-glsp/client": "2.1.0",`

and from 

` "@eclipse-glsp/theia-integration": "2.1.0-next.197"  ==>   "@eclipse-glsp/theia-integration": "2.1.0"`

I can't figure out what the error message is trying to tell me. 

I am using a ShapeView extension that is using `{ Classes, VNode } from 'snabbdom'` and I have a  PolylineEdgeViewWithGapsOnIntersections extension that is also using  `{  VNode } from 'snabbdom' `

Thanks for any hints...

**Comment by tortmayr** [comment, marked as answer]
Hi @rsoika,

that is a strange error. Nothing has changed regarding snappdom between 2.0.0 and 2.1.0, so you should not encounter that Error.
If I recall correctly you previously tested out a locally linked GLSP PR. Did you make sure that all the locally linked packages are 
properly unlinked?  Maybe that is causing the issues. (A brute force way to enforce this would be a full git clean)

**Comment by rsoika** [comment]
Yes this is very strange.
I cleared everything several times with `npm cache clean` and `git clean -xdf`. I also deleted all fragments form `node_modules` and the yarn.lock file. But nothing seemed to help.

Finally I switched my nvm version  with `nvm install 18.19.0` and than I switched back to version `18.17.1`. And this finally did the trick. Now I can compile again.

....I will never ever understand this nodejs and yarn things....

**Comment by tortmayr** [comment]
It seems like this issue can occur due to the dependency version range of snapdom defined in sprotty:
https://github.com/eclipse-sprotty/sprotty/issues/418

If you encounter the problem again I would recommend to use yarn "resolutions" to pin the version to "3.5.1".
